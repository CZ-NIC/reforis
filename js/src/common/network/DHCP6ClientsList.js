/*
 * Copyright (C) 2019-2021 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import PropTypes from "prop-types";
import moment from "moment";
import { formFieldsSize } from "foris";

DHCP6ClientsList.propTypes = {
    ipv6clients: PropTypes.arrayOf(PropTypes.object),
};

export default function DHCP6ClientsList({ ipv6clients }) {
    return (
        <div className={formFieldsSize}>
            <h2>{_("IPv6 DHCP Client List")}</h2>
            <p>
                {_(
                    "This list contains all devices that are connected to the network through wired or wireless connections using IPv6."
                )}
            </p>
            {ipv6clients.length === 0 ? (
                <p className="text-muted text-center">
                    {_("No clients found.")}
                </p>
            ) : (
                <div className="table-responsive">
                    <table className="table table-hover">
                        <thead className="thead-light">
                            <tr className="text-left">
                                <th>{_("Hostname")}</th>
                                <th>{_("IPv6 Address")}</th>
                                <th>{_("DUID")}</th>
                                <th className="text-center">{_("Expires")}</th>
                            </tr>
                        </thead>
                        <tbody>
                            {ipv6clients.map((client) => (
                                <DHCP6ClientsListItem
                                    key={client.ipv6}
                                    {...client}
                                />
                            ))}
                        </tbody>
                    </table>
                </div>
            )}
        </div>
    );
}

DHCP6ClientsListItem.propTypes = {
    ipv6: PropTypes.string.isRequired,
    expires: PropTypes.number.isRequired,
    duid: PropTypes.string.isRequired,
    hostname: PropTypes.string.isRequired,
};

function DHCP6ClientsListItem({ ipv6, expires, duid, hostname }) {
    return (
        <tr className="text-left">
            <td>{hostname}</td>
            <td>{ipv6}</td>
            <td>{duid}</td>
            <td className="text-center">
                {moment.unix(expires).format("YYYY-MM-DD HH:mm")}
            </td>
        </tr>
    );
}
