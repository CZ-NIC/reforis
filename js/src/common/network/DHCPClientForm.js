/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from 'react';
import propTypes from 'prop-types';

import TextInput from '../bootstrap/TextInput';
import {validateDomain} from '../validations';

const HELP_TEXTS = {
    hostname: _('Hostname which will be provided to DHCP server.'),
};

DHCPClientForm.propTypes = {
    formData: propTypes.shape({
        hostname: propTypes.string,
    }).isRequired,
    formErrors: propTypes.shape({
        hostname: propTypes.string
    }),
    setFormValue: propTypes.func.isRequired,
    updateRule: propTypes.func.isRequired,
};

export default function DHCPClientForm({formData, formErrors, setFormValue, updateRule, ...props}) {
    return <TextInput
        label={_('DHCP hostname')}
        value={formData.hostname || ''}
        error={(formErrors || {}).hostname || null}
        helpText={HELP_TEXTS.hostname}

        onChange={setFormValue(
            value => updateRule({hostname: {$set: value}})
        )}

        {...props}
    />
}

export function validateDHCPForm(formData) {
    const error = {hostname: validateDomain(formData.hostname)};
    return error.hostname ? error : null;
}
