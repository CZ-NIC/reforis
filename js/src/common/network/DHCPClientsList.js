/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import PropTypes from "prop-types";
import moment from "moment";

DHCPClientsList.propTypes = {
    clients: PropTypes.arrayOf(PropTypes.object),
};

export default function DHCPClientsList({ clients }) {
    return (
            <h2>{_("DHCP Client List")}</h2>
            <p>
                {_(
                    "This list contains all devices that are connected to the network through wired or wireless connections."
                )}
            </p>
            {clients.length === 0 ? (
                <p className="text-muted text-center">
                    {_("No DHCP clients found.")}
                </p>
            ) : (
                <div className="table-responsive">
                    <table className="table table-hover">
                        <thead>
                            <tr className="text-center">
                                <th>{_("Expires")}</th>
                                <th>{_("IP Address")}</th>
                                <th>{_("MAC Address")}</th>
                                <th>{_("Hostname")}</th>
                                <th>{_("Active")}</th>
                            </tr>
                        </thead>
                        <tbody>
                            {clients.map((client) => (
                                <DHCPClientsListItem
                                    key={client.ip}
                                    {...client}
                                />
                            ))}
                        </tbody>
                    </table>
                </div>
            )}
        </>
    );
}

DHCPClientsListItem.propTypes = {
    ip: PropTypes.string.isRequired,
    expires: PropTypes.number.isRequired,
    mac: PropTypes.string.isRequired,
    hostname: PropTypes.string.isRequired,
    active: PropTypes.bool.isRequired,
};

function DHCPClientsListItem({ ip, expires, mac, hostname, active }) {
    return (
        <tr className="text-center">
            <td>{moment.unix(expires).format("YYYY-MM-DD HH:mm")}</td>
            <td>{ip}</td>
            <td>{mac}</td>
            <td>{hostname}</td>
            <td>
                <i className={`fas ${active ? "fa-check" : "fa-times"}`} />
            </td>
        </tr>
    );
}
