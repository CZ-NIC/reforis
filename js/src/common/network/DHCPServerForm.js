/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from 'react';
import propTypes from 'prop-types';

import NumberInput from 'common/bootstrap/NumberInput';

export const HELP_TEXT = _('Enable this option to automatically assign IP addresses to the devices connected to the router.');

DHCPServerForm.propTypes = {
    formData: propTypes.shape({
        start: propTypes.number,
        limit: propTypes.number,
        lease_time: propTypes.number,
    }).isRequired,
    setFormValue: propTypes.func.isRequired,
    updateRule: propTypes.func.isRequired,
};

export default function DHCPServerForm({formData, updateRule, setFormValue, ...props}) {
    return <>
        <NumberInput
            label={_('DHCP start')}
            value={formData.start}
            min='1'
            required

            onChange={setFormValue(
                value => updateRule({start: {$set: value}})
            )}

            {...props}
        />
        <NumberInput
            label={_('DHCP max leases')}
            value={formData.limit}
            min='1'
            required

            onChange={setFormValue(
                value => updateRule({limit: {$set: value}})
            )}

            {...props}
        />
        <NumberInput
            label={_('Lease time (hours)')}
            value={formData.lease_time}
            min='120'
            required

            onChange={setFormValue(
                value => updateRule({lease_time: {$set: value}})
            )}

            {...props}
        />
    </>
}
