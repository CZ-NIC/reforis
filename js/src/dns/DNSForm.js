/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from 'react';
import CheckBox from '../bootstrap/Checkbox';
import Select from '../bootstrap/Select';
import TextInput from '../bootstrap/TextInput';

const HELP_TEXTS = {
    dns_from_dhcp_enabled: _('This will enable your DNS resolver to place DHCP client\'s names among the local DNS records.'),
    dns_from_dhcp_domain: _('This domain will be used as suffix. E.g. The result for client "android-123" and domain "my.lan" will be "android-123.my.lan".')
};

const DNSSEC_DISABLE_MESSAGE = _(`
DNSSEC is a security technology that protects the DNS communication against attacks on the DNS infrastructure.
We strongly recommend keeping DNSSEC validation enabled unless you know that you will be connecting your device in the
network where DNSSEC is broken. 

Do you still want to continue and stay unprotected?
`);

DNSForm.defaultProps = {
    formErrors: {}
};

export default function DNSForm({formData, formErrors, setFormValue, ...props}) {
    function changeDNSSECHandler(event) {
        if (!event.target.checked && !confirm(DNSSEC_DISABLE_MESSAGE))
            return;
        setFormValue(value => ({dnssec_enabled: {$set: value}}))(event)
    }

    return <>
        <CheckBox
            label={_('Use forwarding')}
            checked={formData.forwarding_enabled}

            onChange={setFormValue(value => ({forwarding_enabled: {$set: value}}))}

            {...props}
        />
        {formData.forwarding_enabled ?
            <Select
                label={_('DNS Forwarder')}
                value={formData.forwarder}
                choices={getForwardersChoices(formData.available_forwarders)}
                onChange={setFormValue(value => ({forwarder: {$set: value}}))}

                {...props}
            />
            : null}
        <CheckBox
            label={_('Enable DNSSEC')}
            checked={formData.dnssec_enabled}

            onChange={changeDNSSECHandler}

            {...props}
        />
        <CheckBox
            label={_('Enable DHCP clients in DNS')}
            checked={formData.dns_from_dhcp_enabled}
            helpText={HELP_TEXTS.dns_from_dhcp_enabled}

            onChange={setFormValue(value => ({dns_from_dhcp_enabled: {$set: value}}))}

            {...props}
        />
        {formData.dns_from_dhcp_enabled ?
            <TextInput
                label={_('Domain of DHCP clients in DNS')}
                value={formData.dns_from_dhcp_domain}
                helpText={HELP_TEXTS.dns_from_dhcp_domain}
                error={formErrors.dns_from_dhcp_domain}

                onChange={setFormValue(value => ({dns_from_dhcp_domain: {$set: value}}))}

                {...props}
            />
            : null}
    </>
}

function getForwardersChoices(available_forwarders) {
    return available_forwarders.reduce(
        (choices, forwarder) => {
            choices [forwarder.name] = forwarder.name === '' ? _('Use provider\'s DNS resolver') : forwarder.description;
            return choices;
        }, {})
}