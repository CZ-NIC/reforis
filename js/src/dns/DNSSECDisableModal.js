/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import PropTypes from "prop-types";

import {
    Modal, ModalBody, ModalFooter, ModalHeader, Button,
} from "foris";

const DNSSEC_DISABLE_MESSAGE = _(`
DNSSEC is a security technology that protects the DNS communication against attacks on the DNS infrastructure.
We strongly recommend keeping DNSSEC validation enabled unless you know that you will be connecting your device in the
network where DNSSEC is broken. 

Do you still want to continue and stay unprotected?
`);

DNSSECDisableModal.propTypes = {
    shown: PropTypes.bool.isRequired,
    setShown: PropTypes.func.isRequired,
    callback: PropTypes.func.isRequired,
};

export default function DNSSECDisableModal({ shown, setShown, callback }) {
    return (
        <Modal setShown={setShown} shown={shown}>
            <ModalHeader setShown={setShown} title={_("Warning!")} />
            <ModalBody><p>{DNSSEC_DISABLE_MESSAGE}</p></ModalBody>
            <ModalFooter>
                <Button onClick={
                    (e) => {
                        e.preventDefault();
                        setShown(false);
                    }
                }
                >
                    {_("Cancel")}
                </Button>
                <Button
                    className="btn-danger"
                    onClick={(e) => {
                        e.preventDefault();
                        callback();
                    }}
                >
                    {_("Disable DNSSEC")}
                </Button>
            </ModalFooter>
        </Modal>
    );
}
