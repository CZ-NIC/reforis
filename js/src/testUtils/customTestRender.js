/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

/* eslint import/export: "off" */

import React from 'react';
import {UIDReset} from 'react-uid';
import {StaticRouter} from 'react-router';
import {render} from '@testing-library/react'

function Wrapper({children}) {
    return <StaticRouter>
        <UIDReset>
            {children}
        </UIDReset>
    </StaticRouter>
}

const customTestRender = (ui, options) => render(ui, {wrapper: Wrapper, ...options});

// re-export everything
export * from '@testing-library/react'

// override render method
export {customTestRender as render}
