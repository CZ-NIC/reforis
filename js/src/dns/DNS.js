/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import PropTypes from "prop-types";

import API_URLs from "common/API";
import { validateDomain } from "common/validations";
import ForisForm from "form/ForisForm";
import ConnectionTest from "connectionTest/ConnectionTest";

import DNSForm from "./DNSForm";

DNS.propTypes = {
    ws: PropTypes.object.isRequired,
    postCallback: PropTypes.func,
};

DNS.defaultProps = {
    postCallback: () => undefined,
};

export default function DNS({ ws, postCallback }) {
    return (
        <>
            <h1>DNS</h1>
            <p dangerouslySetInnerHTML={{
                __html: _(`
Router Turris uses its own DNS resolver with DNSSEC support. It is capable of working independently or it
can forward your DNS queries your internet service provider's DNS resolver.
<br/><br/>
The following setting determines the behavior of the DNS resolver. Usually, it is better to use the ISP's
resolver in networks where it works properly. If it does not work for some reason, it is necessary to use
direct resolving without forwarding.
<br/><br/>
In rare cases ISP's have improperly configured network which interferes with DNSSEC validation. If you
experience problems with DNS, you can <b>temporarily</b> disable DNSSEC validation to determine the source
of the problem. However, keep in mind that without DNSSEC validation, you are vulnerable to DNS spoofing
attacks! Therefore we <b>recommend keeping DNSSEC turned on</b> and resolving the situation with your ISP as
this is a serious flaw on their side.
            `),
            }}
            />
            <ForisForm
                ws={ws}
                forisConfig={{
                    endpoint: API_URLs.dns,
                    wsModule: "dns",
                }}
                postCallback={postCallback}
                validator={validator}
                prepDataToSubmit={prepDataToSubmit}
            >
                <DNSForm />
            </ForisForm>

            <h1>{_("Connection test")}</h1>
            <p
                dangerouslySetInnerHTML={{
                    __html: _(`
Here you can test your internet connection. This test is also useful when you need to check that your DNS resolving 
works as expected. Remember to click on the <b>Save button</b> if you changed your forwarder setting.
        `),
                }}
            />
            <ConnectionTest ws={ws} type="dns" />
        </>
    );
}

function validator(formData) {
    const error = {};
    if (formData.dns_from_dhcp_enabled) {
        error.dns_from_dhcp_domain = validateDomain(formData.dns_from_dhcp_domain);
    }
    return error.dns_from_dhcp_domain ? error : undefined;
}

function prepDataToSubmit(formData) {
    delete formData.available_forwarders;
    if (!formData.forwarding_enabled) delete formData.forwarder;
    return formData;
}
