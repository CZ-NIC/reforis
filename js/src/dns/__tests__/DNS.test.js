/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import diffSnapshot from "snapshot-diff";
import mockAxios from "jest-mock-axios";

import { fireEvent, render, wait, waitForElement } from "foris/testUtils/customTestRender";
import { WebSockets } from "foris";
import { mockJSONError } from "foris/testUtils/network";

import { dnsFixture } from "./__fixtures__/dns";
import { forwardersFixture } from "./__fixtures__/forwarders";

import DNS from "../DNS";

describe("<DNS/>", () => {
    let firstRender;
    let asFragment;
    let getByText;
    let getByLabelText;

    beforeEach(async () => {
        const webSockets = new WebSockets();
        const renderRes = render(<DNS ws={webSockets}/>);
        mockAxios.mockResponse({ data: dnsFixture });
        getByText = renderRes.getByText;
        getByLabelText = renderRes.getByLabelText;
        asFragment = renderRes.asFragment;

        await wait(() => renderRes.getByLabelText(/Use forwarding/));
        firstRender = renderRes.asFragment();
    });

    it("should handle error", async () => {
        const webSockets = new WebSockets();
        const { getByText } = render(<DNS ws={webSockets}/>);
        mockJSONError();
        await wait(() => {
            expect(getByText("An error occurred while fetching data.")).toBeTruthy();
        });
    });

    it("Test with snapshot.", () => {
        expect(firstRender).toMatchSnapshot();
    });

    it("Test with snapshot forwarding.", async () => {
        fireEvent.click(getByText("Use forwarding"));
        mockAxios.mockResponse({ data: forwardersFixture });
        await waitForElement(() => getByLabelText(/Custom forwarder/));
        expect(diffSnapshot(firstRender, asFragment()))
            .toMatchSnapshot();
    });

    it("Test with snapshot DHCP.", () => {
        fireEvent.click(getByText("Enable DHCP clients in DNS"));
        expect(diffSnapshot(firstRender, asFragment()))
            .toMatchSnapshot();
    });

    it("Test post.", () => {
        fireEvent.click(getByText("Use forwarding"));
        fireEvent.click(getByText("Enable DHCP clients in DNS"));
        fireEvent.click(getByText("Save"));

        expect(mockAxios.post)
            .toBeCalled();
        const data = {
            "dns_from_dhcp_domain": "lan",
            "dns_from_dhcp_enabled": true,
            "dnssec_enabled": true,
            "forwarder": "",
            "forwarding_enabled": true
        };
        expect(mockAxios.post)
            .toHaveBeenCalledWith("/reforis/api/dns", data, expect.anything());
    });
});
