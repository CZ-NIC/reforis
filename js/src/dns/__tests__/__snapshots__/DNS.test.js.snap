// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DNS/> Test with snapshot DHCP. 1`] = `
<div>
  <h1>
    DNS
  </h1>
  <p>
    
Router Turris uses its own DNS resolver with DNSSEC support. It is capable of working independently or it
can forward your DNS queries your internet service provider's DNS resolver.

    <br />
    <br />
    
The following setting determines the behavior of the DNS resolver. Usually, it is better to use the ISP's
resolver in networks where it works properly. If it does not work for some reason, it is necessary to use
direct resolving without forwarding.

    <br />
    <br />
    
In rare cases ISP's have improperly configured network which interferes with DNSSEC validation. If you
experience problems with DNS, you can 
    <b>
      temporarily
    </b>
     disable DNSSEC validation to determine the source
of the problem. However, keep in mind that without DNSSEC validation, you are vulnerable to DNS spoofing
attacks! Therefore we 
    <b>
      recommend keeping DNSSEC turned on
    </b>
     and resolving the situation with your ISP as
this is a serious flaw on their side.
            
  </p>
  <form>
    <div
      class="col-sm-12 offset-lg-1 col-lg-10"
      style="margin-bottom: 1rem;"
    >
      <div
        class="custom-control custom-checkbox"
        style="margin-bottom: 0px;"
      >
        <input
          class="custom-control-input"
          id="1"
          type="checkbox"
        />
        <label
          class="custom-control-label"
          for="1"
        >
          Use forwarding
        </label>
      </div>
    </div>
    <div
      class="col-sm-12 offset-lg-1 col-lg-10"
      style="margin-bottom: 1rem;"
    >
      <div
        class="custom-control custom-checkbox"
        style="margin-bottom: 0px;"
      >
        <input
          checked=""
          class="custom-control-input"
          id="2"
          type="checkbox"
        />
        <label
          class="custom-control-label"
          for="2"
        >
          Enable DNSSEC
        </label>
      </div>
    </div>
    <div
      class="col-sm-12 offset-lg-1 col-lg-10"
      style="margin-bottom: 1rem;"
    >
      <div
        class="custom-control custom-checkbox"
        style="margin-bottom: 0px;"
      >
        <input
          class="custom-control-input"
          id="3"
          type="checkbox"
        />
        <label
          class="custom-control-label"
          for="3"
          style="margin-bottom: 0px;"
        >
          Enable DHCP clients in DNS
        </label>
      </div>
      <small
        class="form-text text-muted"
      >
        This will enable your DNS resolver to place DHCP client's names among the local DNS records.
      </small>
    </div>
    <div
      class="col-sm-12 offset-lg-1 col-lg-10"
    >
      <div
        class="form-group"
      >
        <label
          for="7"
        >
          Domain of DHCP clients in DNS
        </label>
        <div
          class="input-group"
        >
          <input
            class="form-control"
            id="7"
            type="text"
            value="lan"
          />
        </div>
        <small
          class="form-text text-muted"
        >
          This domain will be used as suffix. E.g. The result for client "android-123" and domain "my.lan" will be "android-123.my.lan".
        </small>
      </div>
    </div>
    <button
      class="btn btn-primary  offset-lg-8 col-lg-3 col-sm-12"
      type="submit"
    >
       
       
      Save
    </button>
  </form>
  <h1>
    Connection test
  </h1>
  <p>
    
Here you can test your internet connection. This test is also useful when you need to check that your DNS resolving 
works as expected. Remember to click on the 
    <b>
      Save button
    </b>
     if you changed your forwarder setting.
        
  </p>
  <form>
    <button
      class="btn btn-primary offset-lg-8 col-lg-3 col-sm-12"
      type="submit"
    >
       
       
      Test connection
    </button>
  </form>
</div>
`;

exports[`<DNS/> Test with snapshot forwarding. 1`] = `
<div>
  <h1>
    DNS
  </h1>
  <p>
    
Router Turris uses its own DNS resolver with DNSSEC support. It is capable of working independently or it
can forward your DNS queries your internet service provider's DNS resolver.

    <br />
    <br />
    
The following setting determines the behavior of the DNS resolver. Usually, it is better to use the ISP's
resolver in networks where it works properly. If it does not work for some reason, it is necessary to use
direct resolving without forwarding.

    <br />
    <br />
    
In rare cases ISP's have improperly configured network which interferes with DNSSEC validation. If you
experience problems with DNS, you can 
    <b>
      temporarily
    </b>
     disable DNSSEC validation to determine the source
of the problem. However, keep in mind that without DNSSEC validation, you are vulnerable to DNS spoofing
attacks! Therefore we 
    <b>
      recommend keeping DNSSEC turned on
    </b>
     and resolving the situation with your ISP as
this is a serious flaw on their side.
            
  </p>
  <form>
    <div
      class="col-sm-12 offset-lg-1 col-lg-10"
      style="margin-bottom: 1rem;"
    >
      <div
        class="custom-control custom-checkbox"
        style="margin-bottom: 0px;"
      >
        <input
          class="custom-control-input"
          id="1"
          type="checkbox"
        />
        <label
          class="custom-control-label"
          for="1"
        >
          Use forwarding
        </label>
      </div>
    </div>
    <div
      class="form-group col-sm-12 offset-lg-1 col-lg-10"
    >
      <label
        for="5"
      >
        DNS Forwarder
      </label>
      <select
        class="custom-select"
        id="5"
      >
        <option
          value=""
        >
          Use provider's DNS resolver
        </option>
        <option
          value="99_cloudflare"
        >
          Cloudflare (TLS)
        </option>
        <option
          value="99_quad9"
        >
          Quad9 (TLS)
        </option>
        <option
          value="00_odvr-cznic"
        >
          CZ.NIC (TLS)
        </option>
        <option
          value="99_google"
        >
          Google
        </option>
      </select>
    </div>
    <div
      class="col-sm-12 offset-lg-1 col-lg-10"
      style="margin-bottom: 1rem;"
    >
      <div
        class="custom-control custom-checkbox"
        style="margin-bottom: 0px;"
      >
        <input
          checked=""
          class="custom-control-input"
          id="2"
          type="checkbox"
        />
        <label
          class="custom-control-label"
          for="2"
        >
          Enable DNSSEC
        </label>
      </div>
    </div>
    <div
      class="col-sm-12 offset-lg-1 col-lg-10"
      style="margin-bottom: 1rem;"
    >
      <div
        class="custom-control custom-checkbox"
        style="margin-bottom: 0px;"
      >
        <input
          class="custom-control-input"
          id="3"
          type="checkbox"
        />
        <label
          class="custom-control-label"
          for="3"
          style="margin-bottom: 0px;"
        >
          Enable DHCP clients in DNS
        </label>
      </div>
      <small
        class="form-text text-muted"
      >
        This will enable your DNS resolver to place DHCP client's names among the local DNS records.
      </small>
    </div>
    <button
      class="btn btn-primary  offset-lg-8 col-lg-3 col-sm-12"
      type="submit"
    >
       
       
      Save
    </button>
  </form>
  <h1>
    Connection test
  </h1>
  <p>
    
Here you can test your internet connection. This test is also useful when you need to check that your DNS resolving 
works as expected. Remember to click on the 
    <b>
      Save button
    </b>
     if you changed your forwarder setting.
        
  </p>
  <form>
    <button
      class="btn btn-primary offset-lg-8 col-lg-3 col-sm-12"
      type="submit"
    >
       
       
      Test connection
    </button>
  </form>
</div>
`;

exports[`<DNS/> Test with snapshot. 1`] = `
<div>
  <h1>
    DNS
  </h1>
  <p>
    
Router Turris uses its own DNS resolver with DNSSEC support. It is capable of working independently or it
can forward your DNS queries your internet service provider's DNS resolver.

    <br />
    <br />
    
The following setting determines the behavior of the DNS resolver. Usually, it is better to use the ISP's
resolver in networks where it works properly. If it does not work for some reason, it is necessary to use
direct resolving without forwarding.

    <br />
    <br />
    
In rare cases ISP's have improperly configured network which interferes with DNSSEC validation. If you
experience problems with DNS, you can 
    <b>
      temporarily
    </b>
     disable DNSSEC validation to determine the source
of the problem. However, keep in mind that without DNSSEC validation, you are vulnerable to DNS spoofing
attacks! Therefore we 
    <b>
      recommend keeping DNSSEC turned on
    </b>
     and resolving the situation with your ISP as
this is a serious flaw on their side.
            
  </p>
  <form>
    <div
      class="col-sm-12 offset-lg-1 col-lg-10"
      style="margin-bottom: 1rem;"
    >
      <div
        class="custom-control custom-checkbox"
        style="margin-bottom: 0px;"
      >
        <input
          class="custom-control-input"
          id="1"
          type="checkbox"
        />
        <label
          class="custom-control-label"
          for="1"
        >
          Use forwarding
        </label>
      </div>
    </div>
    <div
      class="col-sm-12 offset-lg-1 col-lg-10"
      style="margin-bottom: 1rem;"
    >
      <div
        class="custom-control custom-checkbox"
        style="margin-bottom: 0px;"
      >
        <input
          checked=""
          class="custom-control-input"
          id="2"
          type="checkbox"
        />
        <label
          class="custom-control-label"
          for="2"
        >
          Enable DNSSEC
        </label>
      </div>
    </div>
    <div
      class="col-sm-12 offset-lg-1 col-lg-10"
      style="margin-bottom: 1rem;"
    >
      <div
        class="custom-control custom-checkbox"
        style="margin-bottom: 0px;"
      >
        <input
          class="custom-control-input"
          id="3"
          type="checkbox"
        />
        <label
          class="custom-control-label"
          for="3"
          style="margin-bottom: 0px;"
        >
          Enable DHCP clients in DNS
        </label>
      </div>
      <small
        class="form-text text-muted"
      >
        This will enable your DNS resolver to place DHCP client's names among the local DNS records.
      </small>
    </div>
    <button
      class="btn btn-primary  offset-lg-8 col-lg-3 col-sm-12"
      type="submit"
    >
       
       
      Save
    </button>
  </form>
  <h1>
    Connection test
  </h1>
  <p>
    
Here you can test your internet connection. This test is also useful when you need to check that your DNS resolving 
works as expected. Remember to click on the 
    <b>
      Save button
    </b>
     if you changed your forwarder setting.
        
  </p>
  <form>
    <button
      class="btn btn-primary offset-lg-8 col-lg-3 col-sm-12"
      type="submit"
    >
       
       
      Test connection
    </button>
  </form>
</div>
`;
